manage_breast_screening/record_a_mammogram/views.py

```diff
@@ -115,12 +116,25 @@ def form_valid(self, form):
         if form.cleaned_data["decision"] == "continue":
             return redirect("record_a_mammogram:awaiting_images")
         else:
-            return redirect("record_a_mammogram:appointment_cannot_go_ahead")
-
-
-class AppointmentCannotGoAhead(FormView):
-    template_name = "record_a_mammogram/appointment_cannot_go_ahead.jinja"
-    form_class = AppointmentCannotGoAheadForm
+            return redirect("record_a_mammogram:appointment_cannot_go_ahead", id=self.appointment.pk)
+
+
+def appointment_cannot_go_ahead(request, id):
+    appointment = Appointment.objects.get(pk=id)
+    participant = appointment.screening_episode.participant
+    if request.method == 'POST':
+        form = AppointmentCannotGoAheadForm(request.POST, instance=appointment)
+        if form.is_valid():
+            form.save()
+            return redirect('clinics:index')
+    else:
+        form = AppointmentCannotGoAheadForm(instance=appointment)
```

Comment:
@MatMoore I'm wondering whether it's worth adding some logic here to hydrate the form values from the appointment. In practice, we should be redirecting to the appointment page if the user tries to navigate here after ending the appointment.

Reply:
No, I don't think so. The form doesn't depend on any values from other pages, and we don't direct users to this form if they've already filled it out once. Once you've submitted the form the appointment page just shows a summary with no links to the other pages in the flow.

![Screenshot 2025-05-13 at 15 11 30](https://github.com/user-attachments/assets/7424b49d-d846-48c8-b442-313a2e1c799f)